Docs/beta/_templates/search.html

```diff
@@ -0,0 +1,34 @@
+{% extends "page.html" %}
+
+{%- block regular_scripts -%}
+{{ super() }}
+<script src="{{ pathto('_static/searchtools.js', 1) }}"></script>
+<script src="{{ pathto('_static/language_data.js', 1) }}"></script>
+{%- endblock regular_scripts-%}
+
+{%- block htmltitle -%}
+<title>{{ _("Search") }} - {{ docstitle }}</title>
+{%- endblock htmltitle -%}
+
+{% block content %}
+<noscript>
+<div class="admonition error">
+  <p class="admonition-title">{% trans %}Error{% endtrans %}</p>
+  <p>
+    {% trans %}Please activate JavaScript to enable the search functionality.{% endtrans %}
+  </p>
+</div>
+</noscript>
+
+<div id="search-results"></div>
+{% endblock %}
+
+{% block scripts -%}
+{{ super() }}
+<script src="{{ pathto('searchindex.js', 1) }}"></script>
+{%- endblock scripts %}
+
+{% block site_meta %}
+{{ super() }}
+<meta name="robots" content="noindex" />
+{% endblock site_meta %}
```

Comment:
The search function looks good to me overall, but I find it slightly clumsy.

When I search for "adaround", it shows up as:
![image](https://github.com/user-attachments/assets/67723a0a-05cd-4b37-a522-b7f900b0015c)
and I find it slightly confusing to see the exact same entries in the top 3.
This is because we have three different pages that happen to have same H1 title, one for each of torch, tf, and onnx. I think we have to rename these titles to reflect the framework they are associated with (cc. @quic-hitameht)

@dwelsch-esi  However, as a parallel approach, do you know whether/how we can show the occurrence of the keyword in each page as part of search result?
As a reference, PyTorch API reference will also show you the first occurrence of the keyword in the page
![image](https://github.com/user-attachments/assets/fd6e9c28-b240-4fa5-a377-d3be471c72bd)

Reply:
@dwelsch-esi Looks like this search.html function is already available in the furo theme template, right? If we are reusing the same template, then no point in keeping same file at two places unless we plan to modify the existing files.

Reply:
@quic-kyunggeu I just implemented the default search functionality for the furo theme. There aren't any display options documented, so some investigation and customization would be required if we want to make modifications to it.

@quic-hitameht Apparently the HTML template files need to be in the _templates directory. I tried it without, and the search page was blank. Confusingly, the same is not true of the CSS files in the _static directory. The theme uses the default CSS files in the theme and if there are files in the _static directory (that are designated in the conf.py file) they override the default styles. 

Common practice seems to be to copy the theme's CSS file to the _static directory and make changes to it to customize the theme. My preference is to keep a site-specific CSS file (aimet-furo.css) containing only our changes to the theme, since it's much easier to find and modify styles without all the duplicates in the file.
So yes, we can get rid of the furo.css and furo-extensions.css files from the _static directory. I'll open a PR.